src/App.tsx

```diff
@@ -1,39 +1,66 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export type Props = {};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+export type State = {
+  clockName: string,
+  isClock: boolean,
+};
+
+export class App extends Component<Props, State> {
+  state = {
+    clockName: 'Clock-0',
+    isClock: true,
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.pressedRightMouse);
+
+    document.addEventListener('click', this.pressedLeftmouse);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.pressedRightMouse);
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    document.addEventListener('click', this.pressedLeftmouse);
```

Comment:
you already added listener on componentDidMount, on componentWillUnmount you should remove listeners
```suggestion
    document. removeEventListener('click', this.pressedLeftmouse);
```